prometheus metric relay server part

jfk9w opened the pull request:
Resolves #767 on server side

xtreding commented on server/metrics/prometheus.go:
```diff
@@ -1,28 +1,96 @@
+		Timeout:      timeout,
+	}
+
+	logging.Debugf("✅ Initialized metrics relay: %+v", relay)
```
Since it is an important configuration info - all logs here can have level `info`

jfk9w replied:
These logs are generated by service which should require the least amount of user attention. Since it is also enabled by default, moving the logs to INFO level would draw unwarranted attention (more so in case of problems the user actually has no way or desire to solve, which may also lead to disabling the relay altogether – and we sort of need it).

xtreding commented on server/metrics/relay.go:
```diff
@@ -0,0 +1,132 @@
+}
+
+func (t TickerTrigger) Now() time.Time {
+	return time.Now()
```
Please avoid using `time.Now()` - use custom wrapper instead: `timestamp.Now()`. It's a wrapper for mocking time in tests.

jfk9w replied:
Fixed.

xtreding commented on server/metrics/relay.go:
```diff
@@ -0,0 +1,132 @@
+	ctx, cancel := context.WithCancel(ctx)
+	r.cancel = cancel
+	r.work.Add(1)
+	go func() {
```
Let's use `safego` package for running all goroutines into the safe wrapper which has already had panic handler.

jfk9w replied:
Fixed.

jfk9w commented:
Introduced a metrics relay service which collects Prometheus metrics data and submits it to some HTTP endpoint periodically. The relay is enabled by default and no additional configuration is required.

```yaml
server:
  metrics:
    relay:
      disabled: false # optional, disables relay altogether
      url: https://cplane.jitsu.com/prometheus/relay # optional
      deployment_id: fssGtU # optional, is equal to cluster ID by default
      interval: 1m # optional
      timeout: 1s # optional
```

Metrics data is anonymized by default, but users willing to provide their personalized metrics may set `server.metrics.relay.deployment_id`. Sensitive label values are hashed (currently `source_id` and `destination_id`).

Docker image configuration:
* `JITSU_EXTENTED_TELEMETRY_DISABLED` – same as `server.metrics.relay.disabled`
* `JITSU_EXTENTED_TELEMETRY_DEPLOYMENT_ID` – same as `server.metrics.relay.deployment_id`

The relay sends HTTP POST requests to the specified URL with JSON body of the following structure:
```json
{
    "timestamp": 1623125..., // current time in millis
    "deploymentId": "my_deployment_id", // either cluster ID or deployment_id from configuration if specified
    "hostId": "my-host.domain.com", // hostname as detected by os.Hostname()
    "data": [
        // collected Prometheus data via prometheus.Gatherer.Gather() ([]*dto.MetricFamily)
    ]
}
```

Also:
* Added `source_type`, `source_tap` and `destination_type` labels to relevant Prometheus metrics (in task_executor.go `source_tap` will be empty for now).
